tools/dev/poetrypkg.py

```diff
@@ -198,15 +198,19 @@ def resolve_source_deps(poetry, package, reqs, frozen=False):
 
     from poetry.core.packages.dependency import Dependency
 
-    dep_map = {d['name']: d for d in poetry.locker.lock_data['package']}
+    # normalize deps by lowercasing all the keys
+    dep_map = {d['name'].lower(): d for d in poetry.locker.lock_data['package']}
```

Comment:
e.g. PyYAML vs pyyaml

Reply:
Makes sense, mirroring upstream changes :+1:

Reply:
which upstream changes? this feels very odd, mutating locked package names.

Reply:
There have been a few changes related to package name normalization/canonicalization over time. Specifically, this made me think of the recent change from `pyyaml` --> `PyYAML` in `poetry.lock` you can see [here](https://github.com/cloud-custodian/cloud-custodian/pull/7844/files/89619ee3d9d8797d9ea37288d72b6116d4f74c16#diff-f53a023eedfa3fbf2925ec7dc76eecdc954ea94b7e47065393dbad519613dc89L670-R670):

```  
 233   │  [[package]]
 234   │ -name = "pyyaml"
 235   │ +name = "PyYAML"
 236   │  version = "6.0"
 237   │  description = "YAML parser and emitter for Python"
 238   │  category = "main"
 239   │ @@ -676,7 +676,7 @@ python-versions = ">=3.6"
```